src/include/storage/store/chunked_node_group.h

```diff
@@ -162,6 +173,7 @@ class ChunkedNodeGroup {
     uint64_t capacity;
     std::atomic<common::row_idx_t> numRows;
     std::vector<std::unique_ptr<ColumnChunk>> chunks;
+    std::optional<std::vector<std::unique_ptr<ColumnChunk>>> extraData;
```

Comment:
As we discussed, `ChunkedNodeGroup ` shouldn't be aware of `extraData`, it should just store column chunks based on `columnTypes`, and I think your PR should touch no lines in `ChunkedNodeGroup` ideally.